File: persistenece/src/main/java/com/anokmik/persistence/repository/mock/MockRepository.java
package com.anokmik.persistence.repository.mock;

import com.anokmik.persistence.TripAssistantDatabase;
import com.raizlabs.android.dbflow.config.FlowManager;
import com.raizlabs.android.dbflow.sql.language.SQLite;
import com.raizlabs.android.dbflow.structure.Model;
import com.raizlabs.android.dbflow.structure.database.transaction.FastStoreModelTransaction;

import java.util.List;

public class MockRepository<T extends Model> {

    private final Class<T> type;

    public MockRepository(Class<T> type) {
        this.type = type;
    }

    public void storeModelsFast(List<T> items) {
        FlowManager.getDatabase(TripAssistantDatabase.class).executeTransaction(
                FastStoreModelTransaction.saveBuilder(FlowManager.getModelAdapter(type)).addAll(items).build());
    }

    public long count() {
        return SQLite.select().from(type).count();
    }

}

File: persistenece/src/main/kotlin/com/anokmik/persistence/repository/mock/MockRepository.kt
package com.anokmik.persistence.repository.mock

import com.anokmik.persistence.TripAssistantDatabase
import com.raizlabs.android.dbflow.config.FlowManager
import com.raizlabs.android.dbflow.sql.language.SQLite
import com.raizlabs.android.dbflow.structure.Model
import com.raizlabs.android.dbflow.structure.database.transaction.FastStoreModelTransaction

class MockRepository<in T : Model>(private val type: Class<T>) {

    fun storeModelsFast(items: List<T>) {
        FlowManager.getDatabase(TripAssistantDatabase::class.java).executeTransaction(
                FastStoreModelTransaction.saveBuilder(FlowManager.getModelAdapter(type)).addAll(items).build())
    }

    fun count() = SQLite.select().from(type).count()

}